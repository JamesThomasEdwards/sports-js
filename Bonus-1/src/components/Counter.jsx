// React;
import React from 'react';
// Prop types;
import PropTypes from 'prop-types';

const propTypes = {
    score: PropTypes.number
};
const Counter = (props) => {
    return (
        <div className='counter-container'>
            <p className='counter-container__p'>Counter: {props.score}</p>
        </div>
    );
};

Counter.propTypes = propTypes;

export default Counter;