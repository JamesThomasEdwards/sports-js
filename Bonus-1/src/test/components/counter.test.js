import React from 'react';
import { shallow } from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Counter from '../../components/Counter.jsx';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('should render Counter component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Counter />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});