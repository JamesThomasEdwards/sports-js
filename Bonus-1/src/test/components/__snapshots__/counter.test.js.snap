// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Counter component correctly 1`] = `
<p>
  Counter: 
</p>
`;
